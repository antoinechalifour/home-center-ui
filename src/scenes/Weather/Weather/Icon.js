import React from 'react'
import PropTypes from 'prop-types'
import ClearIcon from 'react-icons/lib/ti/weather-partly-sunny'
import CloudsIcon from 'react-icons/lib/ti/weather-cloudy'
import MistIcon from 'react-icons/lib/ti/waves'
import RainIcon from 'react-icons/lib/ti/weather-shower'

const typeToIcon = {
  clear: ClearIcon,
  clouds: CloudsIcon,
  mist: MistIcon,
  fog: MistIcon,
  rain: RainIcon
}

export default function Icon ({ type }) {
  const WeatherIcon = typeToIcon[type]

  if (!WeatherIcon) {
    console.warn(`Missing icon for weather type "${type}"`)
    return null
  }

  return <WeatherIcon />
}

Icon.propTypes = {
  type: PropTypes.string.isRequired
}
