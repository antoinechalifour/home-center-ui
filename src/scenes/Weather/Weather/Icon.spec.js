jest.mock('react-icons/lib/ti/weather-partly-sunny', () => 'clear')
jest.mock('react-icons/lib/ti/weather-cloudy', () => 'clouds')
jest.mock('react-icons/lib/ti/waves', () => 'mist')
jest.mock('react-icons/lib/ti/weather-shower', () => 'rain')

import React from 'react'
import renderer from 'react-test-renderer'
import Icon from './Icon'

const tests = [
  {
    props: { type: 'clear' }
  },
  {
    props: { type: 'clouds' }
  },
  {
    props: { type: 'mist' }
  },
  {
    props: { type: 'fog' }
  },
  {
    props: { type: 'rain' }
  },
  {
    props: { type: 'cool' }
  }
]

describe('scenes/Weather/Weather/Icon', () => {
  tests.forEach(({ props, expected }) =>
    it(`Should render the correct icon for weather type ${props.type}`, () => {
      const tree = renderer.create(<Icon {...props} />)
      expect(tree).toMatchSnapshot()
    })
  )
})
