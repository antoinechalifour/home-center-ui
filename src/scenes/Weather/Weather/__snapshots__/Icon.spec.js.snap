// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenes/Weather/Weather/Icon Should render the correct icon for weather type clear 1`] = `<clear />`;

exports[`scenes/Weather/Weather/Icon Should render the correct icon for weather type clouds 1`] = `<clouds />`;

exports[`scenes/Weather/Weather/Icon Should render the correct icon for weather type cool 1`] = `null`;

exports[`scenes/Weather/Weather/Icon Should render the correct icon for weather type fog 1`] = `<mist />`;

exports[`scenes/Weather/Weather/Icon Should render the correct icon for weather type mist 1`] = `<mist />`;

exports[`scenes/Weather/Weather/Icon Should render the correct icon for weather type rain 1`] = `<rain />`;
